### PR TITLE
team-list: add cuda team

### PR DESCRIPTION
**Pull request by SomeoneSerge**
###### Description of changes

CC @NixOS/cuda-maintainers 

We can also consider adding

```nix
maintainers = [ ... ] ++ optionals cudaSupport teams.cuda.members;
```

to packages like tensorflow, so that we get notifications for `tensorflowWithCuda` &c. I'm not sure if conditional meta is a particularly bad practice

### CODE REVIEW

File: maintainers/team-list.nix [hunk 1]
```diff
@@ -149,6 +149,19 @@ with lib.maintainers; {
     enableFeatureFreezePing = true;
   };
 
+  cuda = {
+    members = [
+      cigrainger
+      connorbaker
+      knedlsepp
+      samuela
+      SomeoneSerge
```
Comment (SomeoneSerge) [inline maintainers/team-list.nix:158]: Folks, if you'd rather not get personally `@`-ed by the CI, I guess you might want to remove yourself from this list

Comment (RaitoBezarius) [reply maintainers/team-list.nix:158]: Please do not add anyone without their explicit permission.

Comment (RaitoBezarius) [reply maintainers/team-list.nix:158]: Better: let them be added in further PR if they miss the first PR.

File: maintainers/team-list.nix [hunk 2]
```diff
@@ -149,6 +149,19 @@ with lib.maintainers; {
...
+    ];
+    scope = "Maintain CUDA-enabled packages";
+    shortName = "Cuda";
+    githubTeams= [ "cuda-maintainers" ];
```
Comment (RaitoBezarius) [inline maintainers/team-list.nix:162]: formatting issue

File: maintainers/team-list.nix (hunk 1 above)
Comment (SomeoneSerge) [reply maintainers/team-list.nix:158]: Fair, let's have explicit PRs

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**